wazuh/wazuh
# Implement toggle mechanism for Manager Scan in Vulnerability Detection

|Related issue|
|---|
|#22059|

## Objective
This PR introduces a new feature aimed at enhancing the configuration and efficiency of the Vulnerability Detection module by adding an internal option to enable or disable manager scans. By reading a newly introduced parameter from the internal options file, conveying it to the Vulnerability Detection module, and storing this parameter in a RocksDB database, we provide administrators with greater control over the scanning process. This feature includes logic to initiate cleanup or scanning processes based on the parameter's value and its change from the previous state, thereby optimizing the module's operation based on current needs.

## Description
- Implemented functionality to read a new parameter from the internal options file, which indicates whether the manager scan in the Vulnerability Detection module should be enabled or disabled.
- Enhanced the Vulnerability Detection module to accept this parameter and store it in a RocksDB database for persistent state management.
- Added logic to determine the action to be taken based on the current parameter value and its change from the last known state:
  - If the parameter is set to 0 (enabled) and the previous state was 'no' (disabled), a cleanup process is initiated to remove any residuals from previous scans.
  - Conversely, if the last state was 'yes' (enabled) and the parameter is now 1 (disabled), the manager scan is deactivated, and a scanning process is triggered to update the system's vulnerability assessment.
- This approach ensures that changes to the manager scan setting are handled efficiently, maintaining the system's integrity and performance.

## Quality Assurance

### Testing

#### Integration Testing
- Conducted integration testing to assess the overall impact of the new feature on the Vulnerability Detection module's functionality, especially the correct triggering of cleanup or scanning processes based on the internal option's state change.
- **Testing Steps**:
  1. Modify the `managerDisabledScan` in the configuration file inside of `TC-007` to enable/disable the manager scan.
  2. Validate that the Vulnerability Detection module reads this change correctly and updates the RocksDB database accordingly.
  3. Confirm that a cleanup process is triggered when disabling the manager scan from an enabled state, and vice versa, that a scan process is initiated when enabling from a disabled state.

---
